src/main/java/build/buildfarm/cas/CASFileCache.java

```diff
@@ -2140,11 +2148,15 @@ Digest getDigest() {
     }
 
     ListenableFuture<Void> deindexFuture =
-        transform(
+        transformAsync(
             expireFuture,
             result -> {
-              directoriesIndex.remove(digest);
-              return null;
+              try {
+                directoriesIndex.remove(digest);
```

Comment:
what's the purpose of change the transform to async here?

Reply:
To redirect checked IOExceptions into the future. `transform` routines are runnable and may not throw them

Reply:
Good point!